api/v1/db/json/notes-dao.js

```diff
@@ -43,25 +55,17 @@ const filterNotes = (rawNotes, queryParams) => {
  * @returns {Promise} Promise object represents a list of notes
  */
 const getNotes = query => new Promise((resolve, reject) => {
-  // This is the value of the 'source' field that will be set for all notes fetched from the DB.
-  const sourceValue = 'advisorPortal';
-
   try {
-    const { dbDirectoryPath } = config.api;
-    if (!fs.existsSync(dbDirectoryPath)) {
-      reject(new Error(`DB directory path: '${dbDirectoryPath}' is invalid`));
-    }
-
     const { studentID } = query;
     const studentDirPath = `${dbDirectoryPath}/${studentID}`;
     let noteFiles = fs.existsSync(studentDirPath) ? fs.readdirSync(studentDirPath) : [];
     noteFiles = _.filter(noteFiles, it => path.extname(it).toLowerCase() === '.json');
 
     let rawNotes = [];
     _.forEach(noteFiles, (file) => {
-      rawNotes.push(JSON.parse(fs.readFileSync(`${studentDirPath}/${file}`, 'utf8')));
+      rawNotes.push(readJSONFile(`${studentDirPath}/${file}`));
     });
-    _.forEach(rawNotes, (it) => { it.source = sourceValue; });
+    _.forEach(rawNotes, (it) => { it.source = localSourceName; });
```

Comment:
I could be wrong, but it looks like we don't really need two for-loops right? Can we do something like this?

```js
_.forEach(noteFiles, (file) => {
  const rawNote = readJSONFile(`${studentDirPath}/${file}`);
  rawNote.source = localSourceName;
  rawNotes.push(rawNote);
});
```